src/wp-includes/class-wp-block.php

```diff
@@ -171,24 +183,94 @@ public function __construct( $block, $available_context = array(), $registry = n
 	 * value is returned.
 	 *
 	 * @since 5.5.0
+	 * @since 6.6.0 Getting dynamic class properties is deprecated.
 	 *
 	 * @param string $name Property name.
 	 * @return array|null Prepared attributes, or null.
 	 */
 	public function __get( $name ) {
+		if ( 'attributes' !== $name ) {
+			if ( ! static::check_if_public_class_property( $name ) ) {
+				wp_trigger_error(
+					__METHOD__,
+					sprintf( 'Getting the dynamic property "%s" on %s is deprecated.', $name, __CLASS__ ),
+					E_USER_DEPRECATED
+				);
+			}
+			return null;
+		}
+
+		$this->populate_attributes();
+		return $this->attributes;
+	}
+
+
+	/**
+	 * This method returns true for the "attributes" property if it is set, and false otherwise.
+	 * It returns false to reflect that dynamic class properties are not really defined.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string $key Property to check.
+	 * @return bool True if the property exists, false otherwise.
+	 */
+	public function __isset( $name ) {
 		if ( 'attributes' === $name ) {
```

Comment:
@anton-vlasenko before this PR, does `'attributes'` property get called? Is it for backward compatibility? Or is `attributes` actually used within the block editor? I'm asking because it's important to know the reason why it exists and is being called dynamically. Knowing might change the how it gets resolved.

Reply:
https://github.com/WordPress/wordpress-develop/pull/6500

> before this PR, does 'attributes' property get called?

Yes there are several places where the `::$attributes` property is being called. For example:
- block_core_navigation_from_block_get_post_ids(),
- render_block_core_query_pagination_numbers(),
- block_core_post_template_uses_featured_image().

> Is it for backward compatibility? Or is attributes actually used within the block editor?

It was declared because known dynamic properties need to be declared on the class per [this proposal](https://core.trac.wordpress.org/ticket/56034).

> I'm asking because it's important to know the reason why it exists and is being called dynamically. 

It's being called dynamically, but in fact, it's not dynamic (after this PR). If a property is declared on a class and then unset, PHP no longer considers it a dynamic property and does not trigger any deprecation errors.